### PR TITLE
player UI improvements

### LINKED ISSUE #1: Playback
* add more fine tuned controls to control the playback ( forward, backward )
* bar in the bottom?
* overlay buttons?

### PR DESCRIPTION
**Pull request by joonne**
fixes #1 

Add play/pause, forward, backward buttons, make a seek bar on bottom which displays current position as well as duration on the right side.

How does that look? Needs polishing still.

nayttokuva_20180319_001

### CODE REVIEW

**Comment by mlehtima** [comment]
Looks nice. Should be fine after fixing alignment issues in texts and seek bar (red color should not show on the right side of the white circle). Should we use some theme color for the seek bar instead of red?

**Comment by joonne** [comment]
yup, on it :) used the zeropad function to style the duration / position so that looks better now also.

**Comment by joonne** [comment]
nayttokuva_20180320_001

better? :)

**Comment by mlehtima** [comment]
Yes, better, the duration text could be aligned vertically to the center of the seek bar if possible (to me it seems it's slightly below the center). One possibility for the seek bar would have also been the Slider QML element available in Silica reference (with handle and valueText) https://sailfishos.org/develop/docs/silica/qml-sailfishsilica-sailfish-silica-slider.html/. Slider is used for similar use in Gallery app video player. It's just a matter of preferred visual style, either way is ok.

**Comment by joonne** [comment]
Thanks, I totally forgot about that! I'm all in for as native look & feel as possible, I'll refactor a bit.. :)

i.e. preferred visual style is *native* :D

**Comment by joonne** [comment]
nayttokuva_20180320_002

Added native slider & it can be used to navigate in the video as expected.

**Comment by mlehtima** [comment]
You decided not to add the duration in the new version? Otherwise that looks good. Also a few second timeout for automatic hiding of the overlay would be nice to have. Also maybe investigate preventing display blanking during video playback? I think there must be a way to do that.

**Comment by mlehtima** [comment]
https://github.com/skvark/SailKino/blob/master/qml/pages/ScreenBlank.qml could be used to preventing screen blanking. Check for usage in that app.

**Comment by joonne** [comment]
Thanks, good suggestions! :) I thought not to add the duration since native player does not have it either, would you prefer to have it?

I'll add the timeout next & study that blanking prevention.

**Comment by mlehtima** [comment]
We can leave out the duration text for now and if users seem to want it then we can add it later.

**Comment by joonne** [comment]
Should be ready for now.

Screen blanking got resolved easily with that component :)

Added also the timer to hide buttons after some seconds, it gets refreshed upon seeking with buttons or slider.

File: qml/pages/PlayerPage.qml [hunk 1]
```diff
@@ -76,7 +76,10 @@ Page {
 
                 MouseArea {
                     anchors.fill: parent
-                    onClicked: video.seek(video.position - 10000) && overlayTimer.restart()
```
Comment (joonne) [inline qml/pages/PlayerPage.qml]: those functions seem to return falsy values ;S

**Comment by mlehtima** [comment]
For some reason the display blanking prevention kept being enabled even after pressing pause in video player. Otherwise I see no issues. Also should the commits be squashed into one before merging?

File: qml/pages/ScreenBlank.qml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+Item {
+    property bool enabled: false
+
+    function request() {
+        var method = "req_display" + (enabled ? "" : "_cancel") + "_blanking_pause";
+        dbif.call(method, [])
+    }
+
+    onEnabledChanged: enabled && request()
```
Comment (mlehtima) [inline qml/pages/ScreenBlank.qml]: request() is not called if enabled becomes false. This prevents stopping the blanking prevention.

Comment (mlehtima) [reply qml/pages/ScreenBlank.qml]: The display does blank eventually even with this code but it just takes much longer than usual.

**Comment by mlehtima** [comment]
Also a small issue that video playback starts again always after the app becomes visible even if the playback was pause before turning off the display. Not sure if that is intentional.

File: qml/pages/ScreenBlank.qml (hunk 2 above)
Comment (joonne) [reply qml/pages/ScreenBlank.qml]: Oh, I left it running for quite a while and it did not blank, need to test again. :)

Good catch! I did not realize that the request is needed to turn the blanking also off. :sweat:

**Comment by joonne** [comment]
Maybe we should squash, I have never done that before but I guess now is a good time to start learning to do it :)

I left that on purpose as I like it for example when using Netflix app it continues automatically after bringing the app on foreground, that is ofc a bug there but I find it useful that I don't need to tap play after checking a message or anything.

However, not against removing it if it feels as weird behavior and/or a bug :)

**Comment by mlehtima** [comment]
We can leave the current behaviour of resuming the playback and see what users say. Squashing the commits into just one commit should be quite simple with interactive git rebase followed by force pushing to the same branch.

**Comment by mlehtima** [comment]
Looks good now. Display blanking prevention works as excepted. Can be merged.

**Pull request merged**